pkg/catalog/catalog.go

```diff
@@ -415,6 +415,16 @@ func (c *Catalog) CreateBranch(ctx context.Context, repositoryID string, branch
 	if err != nil {
 		return nil, err
 	}
+	if _, err := c.Store.GetCommit(ctx, repository, graveler.CommitID(branchID)); err == nil {
```

Comment:
We can consolidate the duplicate code into a validation function.

Reply:
Done.